R/as_mapper.R

```diff
@@ -62,35 +61,77 @@ as_function <- function(...) {
 
 #' Pluck out a single an element from a vector or environment
 #'
-#' This is a generalised form of `[[` which allows you to index by
-#' name, position, or attribute.
+#' @description
+#'
+#' This is a generalised form of `[[` which allows you to index deeply
+#' and flexibly into data structures. It supports R standard accessors
+#' like integer positions and string names and accepts any accessor
+#' functions.
+#'
+#' `pluck()` is often more readable than a mix of operators and
+#' accessors because it reads linearly and is free of syntactic
+#' cruft. Compare: `accessor(x[[1]])$foo` to `pluck(x, 1, accessor,
+#' "foo")`.
 #'
-#' @param x A vector or environment
-#' @param index A list indexing into the object
-#' @param default Value to use if target is empty or absent.
+#' @param .x A vector or environment
+#' @param ... A list of accessors for indexing into the object. Can be
+#'   an integer position, a string name, or an accessor function.
+#'   These dots [splice lists automatically][rlang::dots_splice]. This
```

Comment:
Probably should also mention that a string will index into the slots of a S4 object and the names of list.